Adds a check for required properties before configuring OAuth flows

bdemers opened the pull request:
Fixes: #119

bdemers commented:
Found an issue with how the `hasProperty` is getting resolved, need to test some testing (and some testing with cloud config to verify a few things)

mraible commented:
The user experience isn't great. It doesn't look like a server is running at all.

<img width="1152" alt="screen shot 2019-03-08 at 10 54 32 am" src="https://user-images.githubusercontent.com/17892/54039338-b0463d00-4190-11e9-8c83-c59413e7bc48.png">

It works great from the command line, but it did before too.

```
$ http localhost:8080
HTTP/1.1 401
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Content-Length: 0
Date: Fri, 08 Mar 2019 15:54:13 GMT
Expires: 0
Pragma: no-cache
Set-Cookie: JSESSIONID=07CD68888EB8C8E84A7DF9DB46C201A7; Path=/; HttpOnly
WWW-Authenticate: Bearer
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
```

bdemers commented:
I agree, this is partly a Chrome issue. Though Safari shows a blank screen so that isn't much better.
Maybe we can, change the default 401 handler? (possibly pushing that back upstream)

bdemers commented on integration-tests/oauth2-reactive/src/main/java/com/okta/spring/tests/oauth2/reactive/implicit/ReactiveImplicitFlowApplication.java:
```diff
@@ -48,8 +49,11 @@ public static void main(String[] args) {
-            return http
-                .authorizeExchange()
+
+            Okta.configureBrowserFriendlyResourceServer401EntryPoint(http);
```
@mraible @dogeared I'm not a big fan of this method name, but I'm stuck, I'm guessing there is something more obvious we could do here.

Anyway, these methods basically make a resource server 401 response go from:
```
HTTP/1.1 401
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Content-Length: 0
Date: Fri, 15 Mar 2019 16:36:18 GMT
Expires: 0
Pragma: no-cache
Set-Cookie: JSESSIONID=some-id; Path=/; HttpOnly
WWW-Authenticate: Bearer
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
```

to:

```
HTTP/1.1 401
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Content-Length: 17
Content-Type: text/plain;charset=ISO-8859-1
Date: Fri, 15 Mar 2019 16:36:18 GMT
Expires: 0
Pragma: no-cache
Set-Cookie: JSESSIONID=some-id; Path=/; HttpOnly
WWW-Authenticate: Bearer
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block

401 UNAUTHORIZED
```

dogeared replied:
`configureResourceServer401ResponseBody`

bdemers commented on integration-tests/oauth2-reactive/src/main/java/com/okta/spring/tests/oauth2/reactive/implicit/ReactiveImplicitFlowApplication.java:
```diff
@@ -48,8 +49,11 @@ public static void main(String[] args) {
-            return http
-                .authorizeExchange()
+
+            Okta.configureResourceServer401ResponseBody(http);
```
@dogeared @mraible any better?

bdemers replied:
re: https://github.com/okta/okta-spring-boot/pull/121/files#r266094025

dogeared replied:
well, since it's my suggestion, I approve! ;)

dogeared commented on integration-tests/oauth2-reactive/src/test/groovy/com/okta/spring/tests/oauth2/implicit/ReactiveImplicitLocalValidationGroupIT.groovy:
```diff
@@ -32,9 +34,23 @@ class ReactiveImplicitLocalValidationGroupIT extends ApplicationTestRunner {
+        .then()
+            .statusCode(401)
+            .header("WWW-Authenticate", startsWith("Bearer"))
+            .body(Matchers.equalTo("401 UNAUTHORIZED"))
```
Can this be a constant? Might change in the future...

dogeared commented on integration-tests/oauth2/src/test/groovy/com/okta/spring/tests/oauth2/implicit/ImplicitLocalValidationGroupIT.groovy:
```diff
@@ -37,4 +39,18 @@ class ImplicitLocalValidationGroupIT extends ApplicationTestRunner {
+        .then()
+            .statusCode(401)
+            .header("WWW-Authenticate", startsWith("Bearer"))
+            .body(Matchers.equalTo("401 UNAUTHORIZED\n"))
```
Why does this one need a newline and the previous test didn't? Maybe a `.contains` with a constant?

bdemers commented:
@dogeared back to you
I cleaned up that text too, instead of using the HttpStatus enum name, i used the reason phrase.  Not a big deal for this the default 401, but it would look nicer for other Exception types.

Removed the extra new line in the body too, now the Reactive and Web outputs are the same
